invertedai/large/common.py

```diff
@@ -0,0 +1,153 @@
+from typing import List, Optional, Tuple
+from pydantic import BaseModel
+
+from invertedai.common import AgentAttributes, AgentState, RecurrentState, Point
+
+class Region(BaseModel):
+    """
+    A data structure to contain information about a region in a map to be used in large simulations.
+
+    See Also
+    --------
+    AgentAttributes
+    """
+
+    center: Optional[Point] = None # The center of the region if such a concept is relevant (e.g. center of a square, center of a rectangle)
+    fov: Optional[float] = None # Side length of the region for the default interpretation of a region as a square
+    agent_states: Optional[List[AgentState]] = None # A list of existing agents within the region
+    agent_attributes: Optional[List[AgentAttributes]] = None # The attributes of agents that exist within the region or that will be initialized within the region
+    recurrent_states: Optional[List[RecurrentState]] = None # Recurrent states of the agents contained within the region
```

Comment:
Is there a semantic difference between `None` and `[]` here? I'm also wondering if we should make it a `List[Tuple[AgentAttributes, AgentState, Optional[RecurrentState]]]` instead, but I'm not convinced.

Reply:
If these Regions were only used for driving then I would agree with the Tuple but since it's used for initialize as well which can be quite wonky, I'd say lets keep state, attributes, and recurrent states separate.